src/types/client.types.ts

```diff
@@ -60,3 +62,7 @@ export interface IClientTokenNone {
 }
 
 export type TClientTokens = IClientTokenNone | IClientTokenBearer | IClientTokenOauth | IClientTokenBasic | IClientOAuth2UserClient;
+
+export interface IClientSettings {
+  agent: Agent;
```

Comment:
[optional] I would write httpAgent
(So people don't think it's related to the userAgent)